tests/Integration/Behaviour/Features/Context/Domain/CustomerFeatureContext.php

```diff
@@ -152,8 +155,114 @@ public function createCustomerUsingCommand(string $customerReference, TableNode
         SharedStorage::getStorage()->set($customerReference, $id->getValue());
     }
+
+    /**
+     * @Transform table:firstName,lastName,email,birthday
+     * @Transform table:firstName,lastName,email,birthday,companyName
+     *
+     * @param TableNode $tableNode
+     *
+     * @return array
+     */
+    public function transformCustomers(TableNode $customersTable): array
+    {
+        return $customersTable->getHash();
+    }
+
+    /**
+     * @When I search for the phrases :searchPhrases I should get the following results:
+     *
+     * @param string $searchPhrases
+     * @param array $expectedCustomers
+     */
+    public function assertFoundCustomers(string $searchPhrases, array $expectedCustomers): void
+    {
+        $foundCustomers = $this->getQueryBus()->handle(new SearchCustomers(explode(' ', $searchPhrases)));
+        $isB2BEnabled = Configuration::get('PS_B2B_ENABLE');
+
+        foreach ($expectedCustomers as $currentExpectedCustomer) {
+            $wasCurrentExpectedCustomerFound = false;
+            foreach ($foundCustomers as $currentFoundCustomer) {
+                if ($currentExpectedCustomer['email'] === $currentFoundCustomer['email']) {
+                    $wasCurrentExpectedCustomerFound = true;
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['firstName'],
+                        $currentFoundCustomer['firstname'],
+                        sprintf(
+                            'Expected and found customers\'s first names don\'t match (%s and %s)',
+                            $currentExpectedCustomer['firstName'],
+                            $currentFoundCustomer['firstname']
+                        )
+                    );
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['lastName'],
+                        $currentFoundCustomer['lastname'],
+                        sprintf(
+                            'Expected and found customers\'s last names don\'t match (%s and %s)',
+                            $currentExpectedCustomer['lastName'],
+                            $currentFoundCustomer['lastname']
+                        )
+                    );
+
+                    Assert::assertEquals(
+                        $currentExpectedCustomer['birthday'],
+                        $currentFoundCustomer['birthday'],
+                        sprintf(
+                            'Expected and found customers\'s birthdays don\'t match (%s and %s)',
+                            $currentExpectedCustomer['birthday'],
+                            $currentFoundCustomer['birthday']
+                        )
+                    );
+
+                    if (!$isB2BEnabled) {
+                        if (isset($currentExpectedCustomer['companyName'])
+                            || isset($currentFoundCustomer['company'])
+                        ) {
+                            throw new RuntimeException(sprintf(
```

Comment:
I like the idea, but what happen if I add a company name, then turn off the B2B setting, and try to save again?
Can you also add this kind of test?

Reply:
Hmm I don't get it...
The case where B2B is disabled is covered, isn't it?

Reply:
@PierreRambaud Can't reproduce it on my side. I'm really not sure to understand what you mean.
When you say "turn off B2B mode", do you refer to the back-office one? Because test is not using the actual configuration value, but the value from the tests environment (set in Given statement) as I understand it.